File: Javascript_DSA/Algorithms/Sorting/selectionSort.js
const swap = require("./helpers");

const selectionSort = (arr) => {
  let minIndex,
    len = arr.length;
  for (let i = 0; i < len; i++) {
    minIndex = i;
    for (let j = i + 1; j < len; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    swap(arr, i, minIndex);
  }
  return arr;
};

selectionSort([7, 5, 2, 4, 3, 9]); //[2, 3, 4, 5, 7, 9]
selectionSort([9, 7, 5, 4, 3, 1]); //[1, 3, 4, 5, 7, 9]
selectionSort([1, 2, 3, 4, 5, 6]); //[1, 2, 3, 4, 5, 6]

File: searchinAndSorting/SelectionSort.c
#include<stdio.h>
void swap(int *xp , int *yp){
	int temp = *xp;
	*xp = *yp;
	*yp = temp;
}

void selectionsort(int arr[],int n){
	int i,j,mid_idx;
	for(i=0;i<n-1;i++)
	{
		mid_idx=i;
		for(j=i+1;j<n;j++)
		{
			if(arr[j]<arr[mid_idx])
			mid_idx=j;
			
		}
		swap(&arr[mid_idx],&arr[i]);
	}
}

void printArray(int arr[],int n){
	printf("Sorted Array. \n");
	for(int i=0;i<n;i++)
	{
		printf("A[%d]=%d",i,arr[i]);
		printf("\n");
	}
}

int main(){
	int arr[]={34,67,32,89,23,70,48};
	int n = sizeof(arr)/sizeof(arr[0]);
	selectionsort(arr,n);
	printArray(arr,n);
	return 0;
	
	
}


















